### ISSUE TITLE
WordSearch is essentially BROKEN!

### ISSUE THREAD

**Issue by 2044295** [evan-evone/WordSearch#3]
So I was trying to use `WordSearch` today during school, and it wasn't working. There were numerous issues, most of which I was able to work around, but long story short, it wasn't finding the words. I'll post the puzzle and the words involved in a little bit, but it's broken and needs to be looked into. I don't know why it wasn't working.

---

Other issues (which will each get their own issues later):

- `WordSearch` couldn't import a puzzle that had accents (é, è, à, etc.)
- `WordSearch` goes onto new lines when printing a puzzle beyond a certain size, despite having 
    plenty of room in the terminal.
- `WordSearch` debug was hard to follow and took up more space than I had available on the screen.

**Comment by 2044295** [comment]
Here is the puzzle:
```
gjmymdlyywgjcclifjmy
wgoreddmewnunepilewn
nzntkltvfloksycaxtyt
ugcleqlhhmusdelouate
nuouulxipeecsutblfde
enugsnlwbrvtivuexirf
acrcjtkebaviollajssi
larhswdmcootsglnqjfl
lrisdeivtirlnsoeulum
unemvtlrnefiyyekpgye
merosuecmipaatqlreyf
etiemtnoielrlpszalna
trdhojntetlamemnllhu
tqpuxnznreripbdsuoet
errzajubuohaxojrpced
rtuihnqqyjmisbsknasl
qnehctocsedbmlnuolnw
uncotontigemowexdekq
weuqitsalenuxnguvdnm
cxdesdiaposymuedqbrd
```

And the words:
```
uneepingle
delacolle
unepellicule
moncourrier
quelsarticles
uneallumette
delouate
devoir
desdiapos
delaficelle
votremonnaie
destimbres
ilmefaut
uncarnet
untrombone
cecolis
delalessive
unepile
quelrayon
uneelastique
descotch
dusopalin
uncotontige
cestvotretour
unstyloabille
```

Yeah, it is in French. I removed the accents (for reason, see above). That shouldn't make a difference though.

For reference, the file (with solution) should be avilable here: [WordSearch](https://drive.google.com/open?id=1CAJ4dmekFOf0cZmvT9LJK4EPeTj68Sfe). The list of words is already translated.

**Comment by 2044295** [comment]
UPDATE: `old.tgz` works! Obviously, the new code is still broken, but `old.tgz` works and could be used to compare the output of the new program. I guess the error is somewhere in the `NumPY` code, we'll need to figure out where it is.

**Comment by evan-evone** [comment]
I'll get to this as soon as I can.
